src/lib/fetcher.ts

```diff
@@ -45,7 +53,47 @@ export default class Fetcher {
     return contentTypes
   }
 
-  async getLocalesForSpace () {
+  async getEditorInterfacesInIntents (intentList: IntentList): Promise<Map<string, APIEditorInterfaces>> {
+    const contentTypeIds: string[] = _.uniq(
+      intentList.getIntents()
+        .filter((intent) => intent.isEditorInterfaceUpdate())
+        .reduce((ids, intent) => {
+          const intentIds = intent.getRelatedContentTypeIds()
+          return ids.concat(intentIds)
+        }, [])
+    )
+
+    let editorInterfaces = new Map<string, APIEditorInterfaces>()
+
+    if (contentTypeIds.length === 0) {
+      return editorInterfaces
+    }
+
+    for (let id of contentTypeIds) {
+      try {
+        const response = await this.makeRequest({
+          method: 'GET',
+          url: `/content_types/${id}/editor_interface`
+        })
+        editorInterfaces.set(id, response)
+      } catch (error) {
+        if (error.name === 'NotFound') {
```

Comment:
We should think about exposing the status code, that might look a bit cleaner than checking against our own API error codes.

Reply:
I'll add a TODO